### PR TITLE
[Fix-689]Add hooks to handle the position of filter

### PR DESCRIPTION
**Pull request by alejandrocabriales**
## Ticket
https://trello.com/c/uFascx9g/689-metrics-line-char-doesnt-show-the-levels-name

## What solved

- [X] Should the section include the level name in all levels.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have performed a self-review of my own chromatic changes
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have checked my code and corrected any misspellings
- [x] I have removed any unnecessary console messages
- [x] I have removed any commented code
- [x] I have checked that there are no buggy stories in Storybook

### CODE REVIEW

File: src/views/Finances/components/SectionPages/ReservesWaterfallChartSection/ReservesWaterfallChartSection.tsx [hunk 1]
```diff
@@ -108,6 +93,8 @@ const ContainerTitleFilter = styled('div')(() => ({
   justifyContent: 'space-between',
   alignItems: 'flex-start',
   rowGap: 8,
+  // border: '2px solid green',
```
Comment (nesdc88) [inline src/views/Finances/components/SectionPages/ReservesWaterfallChartSection/ReservesWaterfallChartSection.tsx:96]: Remove this comment

File: src/core/hooks/useDynamicRelativePosition.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        const containerWidth = containerRef.current.offsetWidth;
+        const titleWidth = titleRef.current.offsetWidth;
+        const filterWidth = filterRef.current.offsetWidth;
+
+        setShouldPositionBelow(titleWidth + filterWidth > containerWidth);
+      }
+    };
+
+    updatePosition();
+    window.addEventListener('resize', updatePosition);
```
Comment (nesdc88) [inline src/core/hooks/useDynamicRelativePosition.tsx:22]: Avoid this if possible, better:

import { useWindowWidth } from '@react-hook/window-size';

then in component:
const width = useWindowWidth();

useEffect(() => {
    const updatePosition = () => {
      if (containerRef.current && titleRef.current && filterRef.current) {
        const containerWidth = containerRef.current.offsetWidth;
        const titleWidth = titleRef.current.offsetWidth;
        const filterWidth = filterRef.current.offsetWidth;

        setShouldPositionBelow(titleWidth + filterWidth > containerWidth);
      }
    };
    updatePosition();
  }, [title, width]);

Comment (nesdc88) [reply src/core/hooks/useDynamicRelativePosition.tsx:22]: This hook: @react-hook/window-size handles resizing in the best possible way.

Comment (alejandrocabriales) [reply src/core/hooks/useDynamicRelativePosition.tsx:22]: Thanks for the feedback, done

File: src/views/Finances/components/SectionPages/ReservesWaterfallChartSection/ReservesWaterfallChartSection.tsx (hunk 1 above)
Comment (alejandrocabriales) [reply src/views/Finances/components/SectionPages/ReservesWaterfallChartSection/ReservesWaterfallChartSection.tsx:96]: done

**Comment by nesdc88** [comment]
LGTM

**Pull request merged**